File: Cpp/Algorithm/src/Chapter3/Merge_88.cpp
#include "../Chapter3/chapter3.h"
void myAlgo::Merge_88::merge(vector<int>& nums1, int m, vector<int>& nums2, int n)
{
	vector<int> tmp;
	for (size_t i = 0; i < m; i++)
	{
		tmp.push_back(nums1[i]);
	}
	for (int i = 0, j = 0; (i + j) < (m + n); )
	{
		if (j == n || (i != m && tmp[i] < nums2[j]))
		{
			nums1[i + j] = tmp[i];
			i++;
		}
		else
		{
			nums1[i + j] = nums2[j];
			j++;
		}
	}
}

File: JavaAlgorithm/src/main/java/algo/tzashinorpu/FourthRound/Chapter03/merge_88.java
package algo.tzashinorpu.FourthRound.Chapter03;

public class merge_88 {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int length = nums1.length;
        int[] nums = new int[length];
        System.arraycopy(nums1, 0, nums, 0, length);
        int i1 = 0;
        int i2 = 0;
        for (int i = 0; i < length; i++) {
            if (i1 < m && i2 < n)
                nums1[i] = nums[i1] < nums2[i2] ? nums[i1++] : nums2[i2++];
            else {
                if (i1 == m)
                    nums1[i] = nums2[i2++];
                else
                    nums1[i] = nums[i1++];
            }
        }
    }
}
